### DISCUSSION TITLE
Create rowspan and editable data in React+Typescript using React table or Any other alternatives?

### DISCUSSION THREAD

**Discussion by zidane-arkan** [TanStack/table#4883] [Q&A]
I am required to make a table according to the provisions above, I have done many ways to be able to make the table above but none of them are suitable, can the experts here provide an alternative in the form of a library or code that can implement the table?
[Table Picture that i want to build here](https://i.stack.imgur.com/JsCT4.png).

[My closest attempt was using the following code](https://codesandbox.io/s/row-span-table-rkq51l?file=/src/Kualifikasi.tsx),
but as you can see. Tables with the same id or number are not combined as in the table image. Even though the existing cells can be edited.

**Comment by talatkuyuk** [comment]
It is **not** related with the `react-table`. It is related with the table html implementation inside the react table component. For every second table row for the same id, you will set the table data cell as `<td rowspan="2">` in the code. 

If your data has 2 records for every id number, and if the id is the first column always; you can implement it like below using the index of the rows and the cells:

```jsx
        <tbody>
          {table.getRowModel().rows.map((row, indexForRow) => (
            <tr key={row.id}>
              {row.getVisibleCells().map((cell, indexForCell) => (
                <td rowspan={indexForRow % 2 === 0 && indexForCell === 0 ? "2" : "1"} key={cell.id}>
                  {flexRender(cell.column.columnDef.cell, cell.getContext())}
                </td>
              ))}
            </tr>
          ))}
        </tbody>
```

You will get the idea...

**Reply by zidane-arkan** [reply]
Thank you for wanting to reply, but I have implemented the code you provided but now the column that was the id is entered into column "0", I have tried deleting the "id" data for series 1 which is in second place, but `<td>` does not disappear and remains in excess

[This is what happen when i try implement your code](https://codesandbox.io/s/row-span-table-rkq51l?file=/src/Kualifikasi.tsx)

**Comment by talatkuyuk** [comment]
Hi, I am not able to give you complete solution, I've tried to explore your mind that there is a `rowSpan` attribute, you can utilize it according to your logic needs. 

If you set a data cell as `<td rowSpan={2}>` in a table row, **you shouldn't provide a correspondent data cell in the next row.**  [Here](https://www.w3schools.com/tags/att_td_rowspan.asp) is an example. Think about that, you can implement your logic according to your data.

**Reply by zidane-arkan** [reply]
Does that mean I have to convert the data form into an array object, if so how do I process it to produce the appropriate table form?

**Reply by talatkuyuk** [reply]
No, you don't have to change the data. The issue is about the implementation of the logic. I gave you some hints, you will find the solution.
